caffe2/python/tutorials/py_gen/MNIST.py

```diff
@@ -128,20 +152,52 @@ def AddInput(model, batch_size, db, db_type):
     return data, label
 
 
-# At this point we need to take a look at the predictions coming out of the network at convert them into probabilities.
-# "What's the probability that this number we're looking at is a 5", or "is this a 7", and so forth. 
+# Now we are going to construct our own model. In this part input is our data blob. The output is a blob which will contain probabilities for each of the 10 possible digits.
+# 
+# We are going to use so called Multilayer Perceptron architecture (MLP). Relu activation function is going to be used:
+# 
+# Relu(x) = x if x > 0 else 0
+# 
+# Each layer of an MLP is just matrix multiplication with a bias + an activation function. I.e.:
+# 
+# layer1 = Relu(X * W1^T + b1)
+# 
+# layer2 = Relu(layer1 * W2^T + b2)
 # 
-# The results will be conformed into a range between 0 and 1 such that the closer you are to 1 the more likely the number matches the prediction. The process that we can use to do this is available in LeNet and will provide us the *softmax* prediction. The `AddLeNetModel` function below will output the `softmax`. However, in this case, it does much more than the softmax - it is the computed model with its convoluted layers, as well as the softmax.
+# ...
 # 
-# [An explanation of kernels in image processing](https://en.wikipedia.org/wiki/Kernel_%28image_processing%29) might be useful for more info on why `kernel=5` is used in the convolutional layers below. `dim_in` is the number of input channels and `dim_out` is the number of output channels. 
+# Then in the end we use Softmax operator to convert scores for each of the digits to probabilities. So (p_0 + ... + p_9) = 1.0 and 0 <= p_i <= 1.0. 
 # 
-# As you can see below `conv1` has 1 channel coming in (`dim_in`) and 20 going out (`dim_out`), whereas `conv2` has 20 going in and 50 going out and `fc3` has 50 going in and 500 going out. The images are transformed to smaller sizes along each convolution.d
+# There are a lot of more detailed MLP explanations online. For example [here](http://deeplearning.net/tutorial/mlp.html).
```

Comment:
There are more detailed MLP explanations online.  A good example is [here](http://deeplearning.net/tutorial/mlp.html).